controller/replica_controller.go

```diff
@@ -244,6 +320,14 @@ func (rc *ReplicaController) syncReplica(key string) (err error) {
 		return rc.ds.RemoveFinalizerForReplica(replica)
 	}
 
+	// Update `Replica.Status.EvictionRequested` field
+	if err = rc.UpdateReplicaEvictionStatus(replica); err != nil {
```

Comment:
I think the defer is to update `replica` from `existingReplica`, since in this separate call is main to update `replica`. And this is before the `ReconcileInstanceState` function call.

Reply:
Sorry. I cannot follow...
The existing replica of the following defer function is used to avoid blindly calling the update function if the replica status is unchanged. And I don't know why you mention the function `ReconcileInstanceState` here. Does that function affect the field update?

Reply:
I think the existing replica and the following defer function is main for [ReconcileInstanceState function](https://github.com/longhorn/longhorn-manager/blob/c5d89d751f7c5d8f78544555fd0ea82581071885/controller/replica_controller.go#L261), and to update `Replica.Status` can before that, that's why I put it before those logic. Please let me know what you think or if you have any other idea.

Reply:
Both `InstanceStatus` and the new field are part of `ReplicaStatus`. And the defer function is dedicated to updating `ReplicaStatus` rather than `InstanceStatus` only.

Reply:
@shuo-wu , I have updated this part, please take a look and let me know. Thanks.